watchmal/dataset/cnn/cnn_dataset.py

```diff
@@ -26,31 +25,40 @@ def __init__(self, h5file, pmt_positions_file, is_distributed, transforms=None,
         super().__init__(h5file, is_distributed)
         
         self.pmt_positions = np.load(pmt_positions_file)['pmt_image_positions']
+        self.use_times = use_times
+        self.use_charges = use_charges
         self.data_size = np.max(self.pmt_positions, axis=0) + 1
         self.barrel_rows = [row for row in range(self.data_size[0]) if
                             np.count_nonzero(self.pmt_positions[:,0] == row) == self.data_size[1]]
-        self.data_size = np.insert(self.data_size, 0, 1)
+        self.data_size = np.insert(self.data_size, 0, 2)
         self.collapse_arrays = collapse_arrays
         self.transforms = None #du.get_transformations(transformations, transforms)
-
-    def process_data(self, hit_pmts, hit_data):
+        
+    def process_data(self, hit_pmts, hit_times, hit_charges):
         """
         Returns event data from dataset associated with a specific index
 
         Args:
             hit_pmts                ... array of ids of hit pmts
-            hid_data                ... array of data associated with hits
+            hit_times               ... array of time data associated with hits
+            hit_charges             ... array of charge data associated with hits
         
         Returns:
             data                    ... array of hits in cnn format
         """
-        hit_pmts -= 1 #SK cable numbers start at 1
+        hit_pmts = hit_pmts-1 #SK cable numbers start at 1
 
         hit_rows = self.pmt_positions[hit_pmts, 0]
         hit_cols = self.pmt_positions[hit_pmts, 1]
 
-        data = np.zeros(self.data_size)
-        data[0, hit_rows, hit_cols] = hit_data
+        data = np.zeros(self.data_size, dtype=np.float32)
+
+        if self.use_times:
+            data[0, hit_rows, hit_cols] = hit_times
+        if self.use_charges:
+            data[1, hit_rows, hit_cols] = hit_charges
```

Comment:
If you're only using either times or charges, the data's first dimension should only be size 1, and the time or charge data goes in 0. If using both, then it should be size 2 then you would have 0 for time and 1 for charge.